pkg/morph/event/subnet/remove_node.go

```diff
@@ -0,0 +1,86 @@
+package subnet
+
+import (
+	"crypto/elliptic"
+	"fmt"
+
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	"github.com/nspcc-dev/neo-go/pkg/rpc/response/result/subscriptions"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neofs-node/pkg/morph/client"
+	"github.com/nspcc-dev/neofs-node/pkg/morph/event"
+	subnetid "github.com/nspcc-dev/neofs-sdk-go/subnet/id"
+)
+
+// RemoveNode structure of subnet.RemoveNode notification from morph chain.
+type RemoveNode struct {
+	subnetID *subnetid.ID
+	nodeKey  *keys.PublicKey
+
+	// txHash is used in notary environmental
+	// for calculating unique but same for
+	// all notification receivers values.
+	txHash util.Uint256
+}
+
+// MorphEvent implements Neo:Morph Event interface.
+func (RemoveNode) MorphEvent() {}
+
+// SubnetworkID returns a marshalled subnetID structure, defined in API.
+func (rn RemoveNode) SubnetworkID() *subnetid.ID { return rn.subnetID }
+
+// Node is public key of the nodeKey that is being deleted.
+func (rn RemoveNode) Node() *keys.PublicKey { return rn.nodeKey }
+
+// TxHash returns hash of the TX with RemoveNode
+// notification.
+func (rn RemoveNode) TxHash() util.Uint256 { return rn.txHash }
+
+const expectedItemNumRemoveNode = 2
+
+// ParseRemoveNode parses notification into subnet event structure.
+//
+// Expects 2 stack items.
+func ParseRemoveNode(e *subscriptions.NotificationEvent) (event.Event, error) {
+	var (
+		ev       RemoveNode
+		rawSubID []byte
+		rawKey   []byte
+		err      error
+	)
+
+	params, err := event.ParseStackArray(e)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse stack items from notify event: %w", err)
+	}
+
+	if ln := len(params); ln != expectedItemNumRemoveNode {
+		return nil, event.WrongNumberOfParameters(expectedItemNumRemoveNode, ln)
+	}
+
+	rawSubID, err = client.BytesFromStackItem(params[0])
+	if err != nil {
+		return nil, fmt.Errorf("could not get raw subnetID: %w", err)
+	}
+
+	ev.subnetID = new(subnetid.ID)
+
+	err = ev.subnetID.Unmarshal(rawSubID)
```

Comment:
Why here? This should be done on IR side. Anyway you don't check if ID refers to zero subnet.

Reply:
deleted marshaling in the parser